src/NUnitFramework/framework/Attributes/TestCaseSourceAttribute.cs

```diff
@@ -122,7 +122,25 @@ public IEnumerable<TestMethod> BuildFrom(IMethodInfo method, Test? suite)
             foreach (TestCaseParameters parms in GetTestCasesFor(method))
             {
                 count++;
-                yield return _builder.BuildTestMethod(method, suite, parms);
+
+                // Clone the parameters to ensure each test method gets its own independent copy
+                // This prevents issues when the same TestCaseData is reused across multiple test methods
+                object?[] clonedArgs = new object?[parms.Arguments.Length];
+                Array.Copy(parms.Arguments, clonedArgs, parms.Arguments.Length);
+
+                var clonedParms = new TestCaseParameters(clonedArgs)
```

Comment:
I didn't know, but the `TestParameters` [constructor](https://github.com/nunit/nunit/blob/main/src/NUnitFramework/framework/Internal/TestParameters.cs#L66) already copies the arguments. In addition, in the place where we DO modify the parameters we make another copy.